grakn-factory/titan-factory/src/main/java/ai/grakn/graph/internal/GraknTitanGraph.java

```diff
@@ -90,8 +91,25 @@ protected void clearGraph() {
 
     @Override
     public void commitTransactionInternal(){
+        executeLockingMethod(super::commitTransactionInternal);
+    }
+
+    @Override
+    Vertex addVertex(Schema.BaseType baseType){
+        final Vertex[] v = new Vertex[1];
+        executeLockingMethod(() -> v[0] = super.addVertex(baseType));
+        return v[0];
```

Comment:
Thought of that. The problem is that sometimes the function passed into `executeLockingMethod` does return something, sometimes it doesn't. 

I thought of using an optional in that case but it still looks ugly. Either way you end up either returning a null which gets ignored (not ideal I admit) or this ugly thing.

Any preference on the ugliness ?